### ISSUE TITLE
IsolationForestModel  defaultParamMap has params that do not exist[BUG]

### ISSUE THREAD

**Issue by mariinash** [microsoft/SynapseML#1587]
### SynapseML version

synapseml: 0.10.0

### System information

- **Language version** python 3.7, scala 2.12
- **Spark Version** 3.2.2
- **Spark Platform** Local (Docker)

### Describe the problem

If I try to write a very simple IsolationForest, I can't transform the fitted model due to params in DefaultParamMap that do not exist. Moreover, I have not been able to get the attributes the model was fitted with, problems which were fixed in [https://github.com/microsoft/SynapseML/pull/1330](url).

### Code to reproduce issue

from synapse.ml.isolationforest import *

isolationForest = IsolationForest()
                           .setNumEstimators(40)
                           .setBootstrap(False)
                           .setFeaturesCol("features")
                           .setPredictionCol("predictedLabel")
                           .setScoreCol("outlierScore")
                           .setRandomSeed(1)

model = isolationForest.fit(X)

print(model.__dict__) # ----> This gives:

'''
{'_input_kwargs': {'java_obj': JavaObject id=o673}, 'uid': 'IsolationForestModel_fdc4dfe11ce1', '_paramMap': {Param(parent='IsolationForestModel_fdc4dfe11ce1', name='innerModel', doc='the fit isolation forrest instance'): JavaObject id=o677}, '_defaultParamMap': {Param(parent='IsolationForestModel_fdc4dfe11ce1', name='bootstrap', doc='If true, draw sample for each tree with replacement. If false, do not sample with replacement.'): False, Param(parent='IsolationForestModel_fdc4dfe11ce1', name='contamination', doc='The fraction of outliers in the training data set. If this is set to 0.0, it speeds up the training and all predicted labels will be false. The model and outlier scores are otherwise unaffected by this parameter.'): 0.0, Param(parent='IsolationForestModel_fdc4dfe11ce1', name='contaminationError', doc='The error allowed when calculating the threshold required to achieve the specified contamination fraction. The default is 0.0, which forces an exact calculation of the threshold. The exact calculation is slow and can fail for large datasets. If there are issues with the exact calculation, a good choice for this parameter is often 1% of the specified contamination value.'): 0.0, Param(parent='IsolationForestModel_fdc4dfe11ce1', name='featuresCol', doc='The feature vector.'): 'features', Param(parent='IsolationForestModel_fdc4dfe11ce1', name='maxFeatures', doc='The number of features used to train each tree. If this value is between 0.0 and 1.0, then it is treated as a fraction. If it is >1.0, then it is treated as a count.'): 1.0, Param(parent='IsolationForestModel_fdc4dfe11ce1', name='maxSamples', doc='The number of samples used to train each tree. If this value is between 0.0 and 1.0, then it is treated as a fraction. If it is >1.0, then it is treated as a count.'): 256.0, Param(parent='IsolationForestModel_fdc4dfe11ce1', name='numEstimators', doc='The number of trees in the ensemble.'): 100, Param(parent='IsolationForestModel_fdc4dfe11ce1', name='predictionCol', doc='The predicted label.'): 'predictedLabel', Param(parent='IsolationForestModel_fdc4dfe11ce1', name='randomSeed', doc='The seed used for the random number generator.'): 1, Param(parent='IsolationForestModel_fdc4dfe11ce1', name='scoreCol', doc='The outlier score.'): 'outlierScore'}, '_params': [Param(parent='IsolationForestModel_fdc4dfe11ce1', name='bootstrap', doc='If true, draw sample for each tree with replacement. If false, do not sample with replacement.'), Param(parent='IsolationForestModel_fdc4dfe11ce1', name='contamination', doc='The fraction of outliers in the training data set. If this is set to 0.0, it speeds up the training and all predicted labels will be false. The model and outlier scores are otherwise unaffected by this parameter.'), Param(parent='IsolationForestModel_fdc4dfe11ce1', name='contaminationError', doc='The error allowed when calculating the threshold required to achieve the specified contamination fraction. The default is 0.0, which forces an exact calculation of the threshold. The exact calculation is slow and can fail for large datasets. If there are issues with the exact calculation, a good choice for this parameter is often 1% of the specified contamination value.'), Param(parent='IsolationForestModel_fdc4dfe11ce1', name='featuresCol', doc='The feature vector.'), Param(parent='IsolationForestModel_fdc4dfe11ce1', name='innerModel', doc='the fit isolation forrest instance'), Param(parent='IsolationForestModel_fdc4dfe11ce1', name='maxFeatures', doc='The number of features used to train each tree. If this value is between 0.0 and 1.0, then it is treated as a fraction. If it is >1.0, then it is treated as a count.'), Param(parent='IsolationForestModel_fdc4dfe11ce1', name='maxSamples', doc='The number of samples used to train each tree. If this value is between 0.0 and 1.0, then it is treated as a fraction. If it is >1.0, then it is treated as a count.'), Param(parent='IsolationForestModel_fdc4dfe11ce1', name='numEstimators', doc='The number of trees in the ensemble.'), Param(parent='IsolationForestModel_fdc4dfe11ce1', name='predictionCol', doc='The predicted label.'), Param(parent='IsolationForestModel_fdc4dfe11ce1', name='randomSeed', doc='The seed used for the random number generator.'), Param(parent='IsolationForestModel_fdc4dfe11ce1', name='scoreCol', doc='The outlier score.')], 'bootstrap': Param(parent='IsolationForestModel_fdc4dfe11ce1', name='bootstrap', doc='If true, draw sample for each tree with replacement. If false, do not sample with replacement.'), 'contamination': Param(parent='IsolationForestModel_fdc4dfe11ce1', name='contamination', doc='The fraction of outliers in the training data set. If this is set to 0.0, it speeds up the training and all predicted labels will be false. The model and outlier scores are otherwise unaffected by this parameter.'), 'contaminationError': Param(parent='IsolationForestModel_fdc4dfe11ce1', name='contaminationError', doc='The error allowed when calculating the threshold required to achieve the specified contamination fraction. The default is 0.0, which forces an exact calculation of the threshold. The exact calculation is slow and can fail for large datasets. If there are issues with the exact calculation, a good choice for this parameter is often 1% of the specified contamination value.'), 'featuresCol': Param(parent='IsolationForestModel_fdc4dfe11ce1', name='featuresCol', doc='The feature vector.'), 'innerModel': Param(parent='IsolationForestModel_fdc4dfe11ce1', name='innerModel', doc='the fit isolation forrest instance'), 'maxFeatures': Param(parent='IsolationForestModel_fdc4dfe11ce1', name='maxFeatures', doc='The number of features used to train each tree. If this value is between 0.0 and 1.0, then it is treated as a fraction. If it is >1.0, then it is treated as a count.'), 'maxSamples': Param(parent='IsolationForestModel_fdc4dfe11ce1', name='maxSamples', doc='The number of samples used to train each tree. If this value is between 0.0 and 1.0, then it is treated as a fraction. If it is >1.0, then it is treated as a count.'), 'numEstimators': Param(parent='IsolationForestModel_fdc4dfe11ce1', name='numEstimators', doc='The number of trees in the ensemble.'), 'predictionCol': Param(parent='IsolationForestModel_fdc4dfe11ce1', name='predictionCol', doc='The predicted label.'), 'randomSeed': Param(parent='IsolationForestModel_fdc4dfe11ce1', name='randomSeed', doc='The seed used for the random number generator.'), 'scoreCol': Param(parent='IsolationForestModel_fdc4dfe11ce1', name='scoreCol', doc='The outlier score.'), '_java_obj': JavaObject id=o673}
'''

print(model.getInnerModel().getNumEstimators()) # ----> 100 (default one)

df = model.transform(X)

### Other info / logs

py4j.protocol.Py4JJavaError: An error occurred while calling o673.getParam.
 : java.util.NoSuchElementException: Param bootstrap does not exist.
at org.apache.spark.ml.param.Params.$anonfun$getParam$2(params.scala:705)
at scala.Option.getOrElse(Option.scala:189)
at org.apache.spark.ml.param.Params.getParam(params.scala:705)
at org.apache.spark.ml.param.Params.getParam$(params.scala:703)
at org.apache.spark.ml.PipelineStage.getParam(Pipeline.scala:41)
at sun.reflect.GeneratedMethodAccessor64.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
at py4j.Gateway.invoke(Gateway.java:282)
at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
at py4j.commands.CallCommand.execute(CallCommand.java:79)
at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
at java.lang.Thread.run(Thread.java:748)

If I set defaultParamMap as an empty dict {}, it works, but I still cannot get the parameters I trained the model with.

### What component(s) does this bug affect?

- [ ] `area/cognitive`: Cognitive project
- [X] `area/core`: Core project
- [ ] `area/deep-learning`: DeepLearning project
- [ ] `area/lightgbm`: Lightgbm project
- [ ] `area/opencv`: Opencv project
- [ ] `area/vw`: VW project
- [ ] `area/website`: Website
- [ ] `area/build`: Project build system
- [ ] `area/notebooks`: Samples under notebooks folder
- [ ] `area/docker`: Docker usage
- [ ] `area/models`: models related issue

### What language(s) does this bug affect?

- [ ] `language/scala`: Scala source code
- [X] `language/python`: Pyspark APIs
- [ ] `language/r`: R APIs
- [ ] `language/csharp`: .NET APIs
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [X] `integrations/synapse`: Azure Synapse integrations
- [ ] `integrations/azureml`: Azure ML integrations
- [ ] `integrations/databricks`: Databricks integrations

AB#1899264

**Comment by mariinash** [comment]
Nevermind, I found the issue.
